types/npm-package-arg/index.d.ts

```diff
@@ -15,34 +15,9 @@ declare namespace npa {
      * Something like: 1.2, ^1.7.17, http://x.com/foo.tgz, git+https://github.com/user/foo, bitbucket:user/foo, file:foo.tar.gz or file:../foo/bar/. If not included then the default is latest.
      * @param where Optionally the path to resolve file paths relative to. Defaults to process.cwd()
      */
-    function resolve(name: string, spec: string, where?: string):
-        | FileResult
-        | HostedGitResult
-        | URLResult
-        | AliasResult
-        | RegistryResult;
-
-    class Result {
-        /**
-         * One of the following strings:
-         * * git - A git repo
-         * * tag - A tagged version, like "foo@latest"
-         * * version - A specific version number, like "foo@1.2.3"
-         * * range - A version range, like "foo@2.x"
-         * * file - A local .tar.gz, .tar or .tgz file.
-         * * directory - A local directory.
-         * * remote - An http url (presumably to a tgz)
-         */
-        type:
-            | "git"
-            | "tag"
-            | "version"
-            | "range"
-            | "file"
-            | "directory"
-            | "remote"
-            | "alias";
+    function resolve(name: string, spec: string, where?: string): Result;
 
+    class BaseResult {
```

Comment:
You cannot rename `Result` to `BaseResult` as `Result` has an actual [runtime definition](https://github.com/npm/npm-package-arg/blob/74d06ae66df6b8a91f5b5e4b8ff001a887cf4851/lib/npa.js#L130), while `BaseResult` does not.